### PR TITLE
framework: use the macro instead

### PR DESCRIPTION
**Pull request by mickeyouyou**

### CODE REVIEW

File: cyber/base/unbounded_queue.h [hunk 1]
```diff
@@ -71,9 +71,7 @@ class UnboundedQueue {
     do {
       head_next = old_head->next;
 
-      if (head_next == nullptr) {
-        return false;
-      }
+      RETURN_VAL_IF_NULL(head_next, false);
```
Comment (xiaoxq) [inline cyber/base/unbounded_queue.h]: Personally I'm against macros. See a long story here:
https://github.com/ApolloAuto/apollo/pull/7552#issuecomment-478438061

I'll leave the decision to Cyber owners.

Comment (GoLancer) [reply cyber/base/unbounded_queue.h]: Don't use such macros here, especially it requires extra dependency.

File: cyber/tools/cyber_recorder/recorder.cc [hunk 2]
```diff
@@ -193,10 +187,7 @@ void Recorder::ReaderCallback(const std::shared_ptr<RawMessage>& message,
     return;
   }
 
-  if (message == nullptr) {
-    AERROR << "message is nullptr, channel: " << channel_name;
```
Comment (fengqikai1414) [inline cyber/tools/cyber_recorder/recorder.cc]: The log needs to be kept

File: cyber/transport/dispatcher/rtps_dispatcher.cc [hunk 3]
```diff
@@ -40,10 +40,7 @@ void RtpsDispatcher::Shutdown() {
 }
 
 void RtpsDispatcher::AddSubscriber(const RoleAttributes& self_attr) {
-  if (participant_ == nullptr) {
-    AWARN << "please set participant firstly.";
```
Comment (fengqikai1414) [inline cyber/transport/dispatcher/rtps_dispatcher.cc]: The log needs to be kept

File: cyber/tools/cyber_recorder/recorder.cc [hunk 4]
```diff
@@ -174,10 +171,7 @@ bool Recorder::InitReaderImpl(const std::string& channel_name,
     config.pending_queue_size =
         gflags::Int32FromEnv("CYBER_PENDING_QUEUE_SIZE", 50);
     reader = node_->CreateReader<RawMessage>(config, callback);
-    if (reader == nullptr) {
-      AERROR << "Create reader failed.";
```
Comment (fengqikai1414) [inline cyber/tools/cyber_recorder/recorder.cc]: The log needs to be kept

File: cyber/tools/cyber_recorder/recorder.cc [hunk 5]
```diff
@@ -59,10 +59,7 @@ bool Recorder::Start() {
   is_started_ = true;
   display_thread_ =
       std::make_shared<std::thread>([this]() { this->ShowProgress(); });
-  if (display_thread_ == nullptr) {
-    AERROR << "init display thread error.";
```
Comment (fengqikai1414) [inline cyber/tools/cyber_recorder/recorder.cc]: The log needs to be kept

File: cyber/tools/cyber_recorder/player/play_task_producer.cc [hunk 6]
```diff
@@ -189,10 +189,7 @@ bool PlayTaskProducer::UpdatePlayParam() {
 bool PlayTaskProducer::CreateWriters() {
   std::string node_name = "cyber_recorder_play_" + std::to_string(getpid());
   node_ = apollo::cyber::CreateNode(node_name);
-  if (node_ == nullptr) {
-    AERROR << "create node failed.";
```
Comment (fengqikai1414) [inline cyber/tools/cyber_recorder/player/play_task_producer.cc]: The log needs to be kept

File: cyber/tools/cyber_recorder/player/play_task.cc [hunk 7]
```diff
@@ -32,10 +32,7 @@ PlayTask::PlayTask(const MessagePtr& msg, const WriterPtr& writer,
       msg_play_time_ns_(msg_play_time_ns) {}
 
 void PlayTask::Play() {
-  if (writer_ == nullptr) {
-    AERROR << "writer is nullptr, can't write message.";
```
Comment (fengqikai1414) [inline cyber/tools/cyber_recorder/player/play_task.cc]: The log needs to be kept

File: cyber/record/file/record_file_writer.cc [hunk 8]
```diff
@@ -46,10 +46,7 @@ bool RecordFileWriter::Open(const std::string& path) {
   chunk_flush_.reset(new Chunk());
   is_writing_ = true;
   flush_thread_ = std::make_shared<std::thread>([this]() { this->Flush(); });
-  if (flush_thread_ == nullptr) {
-    AERROR << "Init flush thread error.";
```
Comment (fengqikai1414) [inline cyber/record/file/record_file_writer.cc]: The log needs to be kept

File: cyber/python/examples/py_service.cc [hunk 9]
```diff
@@ -30,10 +30,7 @@ using apollo::cyber::proto::Chatter;
 using apollo::cyber::message::PyMessageWrap;
 apollo::cyber::PyService *server = nullptr;
 int service_callback(const char *channel_name) {
-  if (server == nullptr) {
-    AERROR << "server is null.";
```
Comment (fengqikai1414) [inline cyber/python/examples/py_service.cc]: The log needs to be kept

File: cyber/py_wrapper/py_node.h [hunk 10]
```diff
@@ -374,10 +362,7 @@ class PyChannelUtils {
 
   static std::string get_msgtype_by_channelname(const std::string& channel_name,
                                                 uint8_t sleep_s = 0) {
-    if (channel_name.empty()) {
-      AERROR << "channel_name is null";
```
Comment (fengqikai1414) [inline cyber/py_wrapper/py_node.h]: The log needs to be kept

File: cyber/base/unbounded_queue.h (hunk 1 above)
Comment (freeHackOfJeff) [reply cyber/base/unbounded_queue.h]: Btw, please don't use framework as the commits title tag, use cyber instead.

Comment (mickeyouyou) [reply cyber/base/unbounded_queue.h]: thanks @xiaoxq and @GoLancer. revert all changes in directory `cyber/base`

File: cyber/tools/cyber_recorder/recorder.cc (hunk 2 above)
Comment (mickeyouyou) [reply cyber/tools/cyber_recorder/recorder.cc]: fixed.

**Pull request closed without merging**